internal/proxy/request_signer.go

```diff
@@ -0,0 +1,176 @@
+package proxy
+
+import (
+	"bytes"
+	"crypto"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/sha256"
+	"crypto/x509"
+	"encoding/base64"
+	"encoding/pem"
+	"fmt"
+	"hash"
+	"io/ioutil"
+	"net/http"
+	"strings"
+)
+
+// Only headers enumerated in this list are used to compute the signature of a request.
+var signedHeaders = []string{
+	"Content-Length",
+	"Content-Md5",
+	"Content-Type",
+	"Date",
+	"Authorization",
+	"X-Forwarded-User",
+	"X-Forwarded-Email",
+	"X-Forwarded-Groups",
+	"Cookie",
+}
+
+// Name of the header used to transmit the signature computed for the request.
+var signatureHeader = "Octoboi-Signature"
```

Comment:
I hate to be a buzzkill, but let's do something a little more professional here. Maybe `SSO-Signature` or even just `Signature`